massgen/tests/test_cb_adaptive.py

```
@@ -0,0 +1,706 @@
+
+
+class TestConcurrency:
+    def test_concurrent_record_outcome_is_thread_safe(self, controller):
+        """8 threads x 1000 record_outcome calls must produce valid EWMA in [0, 1]."""
+        errors: list[Exception] = []
+
+        def worker(is_failure: bool) -> None:
+            try:
+                for _ in range(1000):
+                    controller.record_outcome(is_failure)
+            except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)